_rules/form-field-visual-label-is-descriptive-1cd4da.md

```diff
@@ -0,0 +1,211 @@
+
+## Expectation
+
+The [visible][] `label` element describes the purpose of the associated form field element. The [visual context][] of the element can be used to differentiate the purpose from other form fields on the same page.
+
+**Note:** In this case, [visual context][] can be created by headings, field sets and legends, text that is near the control, etc.
+
+**Note**: Labels do not need to be lengthy. A word, or even a single character, may suffice.
+
+## Assumptions
+
+Having a differentiating [context][] that is not [programmatically determinable](https://www.w3.org/TR/WCAG21/#dfn-programmatically-determinable) might be a violation under other [WCAG](https://www.w3.org/TR/WCAG21/) success criteria. For example, using text that has not been marked up as headings in order to split a form into different sections is a violation of [success criterion 1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships). This rule assumes, however, that this is allowed under [success criterion 2.4.6 Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels).
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.4.6: Headings and Labels](https://www.w3.org/WAI/WCAG21/Understanding/headings-and-labels.html)
+- [G131: Providing descriptive labels](https://www.w3.org/WAI/WCAG21/Techniques/general/G131)
+- [H44: Using label elements to associate text labels with form controls](https://www.w3.org/WAI/WCAG21/Techniques/html/H44)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [visible][] `label` element describes the purpose of the associated element.
+
+```html
+<label for="fname">First name:</label> <input id="fname" type="text" name="fname" />
+```
+
+#### Passed Example 2
+
+The implicit label, that is coded with the `label` element, describes the purpose of the associated element.
+
+```html
+<label>First name:<input id="fname" type="text" name="fname"/></label>
+```
+
+#### Passed Example 3
+
+The headings provide a [visual context][] that differentiates the purpose of the otherwise identically named form fields, [visual context][] does not need to be [included in the accessibility tree][].
+
+```html
+<h2 aria-hidden="true">Shipping address</h2>
+<label>Name<input id="shipping-name" type="text" name="name"/></label>
+<label>Street<input id="shipping-street" type="text" name="street"/></label>
+
+<h2 aria-hidden="true">Billing address</h2>
+<label>Name<input id="billing-name" type="text" name="name"/></label>
+<label>Street<input id="billing-street" type="text" name="street"/></label>
+```
+
+#### Passed Example 4
+
+The [visible][] `label` element describes the form field.
+
+```html
+<label for="fname">First name:</label> <input id="fname" type="text" name="fname" aria-label="Fill in:" />
```

Comment:
```html
<label for="fname">First name:</label> <input id="fname" type="text" name="fname" aria-label="Fill in:" /> 
````
This example is passed for the rule "Form field visual label is descriptive" . 

But it's a failed one for the rule "Form field visual label is descriptive" (failed 7)

Since those 2 rules are evaluating the same SC (2.4.6) we can't have same test cases that leads to different outcomes (and failed vs passsed). 

And I thought we agree, per this issue #546, on the fact that **test cases need to meet the accessibility requirement** ? 

If so, then some test cases need to be changed? 

````html
<label for="fname">First name:</label> <input id="fname" type="text" name="fname" aria-label="Fill in:" />
````

should be : 

````html
<label for="fname">First name:</label> <input id="fname" type="text" name="fname" aria-label="First name:" />
`````